ush/python/pygfs/task/jedi.py

```diff
@@ -0,0 +1,276 @@
+            self.config = parse_j2yaml(task_config.JEDIYAML, task_config,
+                                       searchpath=self.j2tmpl_dir)
+        else:
+            raise KeyError(f"Task config must contain JCB_BASE_YAML or JEDIYAML")
+
+        logger.debug(f"JEDI config:\n{pformat(self.config)}")
+
+    @logit(logger)
+    def execute(self, task_config: AttrDict[str, Any], aprun_cmd: str, jedi_args: Optional[List] = None) -> None:
+        """Execute JEDI application
+
+        Parameters
+        ----------
+        task_config : AttrDict
+            Dictionary of all configuration variables associated with a GDAS task.
+        aprun_cmd: str
+            String comprising the run command for the JEDI executable.
+        jedi_args (optional): List
+            List of strings comprising optional input arguments for the JEDI executable.
+
+        Returns
+        ----------
+        None
+        """
+
+        chdir(task_config.DATA)
+
+        exec_cmd = Executable(aprun_cmd)
+        exec_cmd.add_default_arg(self.exe)
+        if jedi_args:
+            for arg in jedi_args:
+                exec_cmd.add_default_arg(arg)
+        exec_cmd.add_default_arg(self.yaml)
+
+        try:
+            logger.debug(f"Executing {exec_cmd}")
+            exec_cmd()
+        except OSError:
+            raise OSError(f"Failed to execute {exec_cmd}")
+        except Exception:
+            raise WorkflowException(f"An error occured during execution of {exec_cmd}")
+
+        pass
+
+    def link_exe(self, task_config: AttrDict[str, Any]) -> None:
+        """Link JEDI executable to run directory
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        ----------
+        None
+        """
+
+        # TODO: linking is not permitted per EE2.  Needs work in JEDI to be able to copy the exec.
+        logger.info(f"Link executable {task_config.JEDIEXE} to DATA/")
+        logger.warn("Linking is not permitted per EE2.")
+        exe_dest = os.path.join(task_config.DATA, os.path.basename(task_config.JEDIEXE))
+        if os.path.exists(exe_dest):
+            rm_p(exe_dest)
+        os.symlink(task_config.JEDIEXE, exe_dest)
+
+    @logit(logger)
+    def get_obs_dict(self, task_config: AttrDict[str, Any]) -> Dict[str, Any]:
+        """Compile a dictionary of observation files to copy
+
+        This method extracts 'observers' from the JEDI yaml and from that list, extracts a list of
+        observation files that are to be copied to the run directory
+        from the observation input directory
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        ----------
+        obs_dict: Dict
+            a dictionary containing the list of observation files to copy for FileHandler
+        """
+
+        logger.info(f"Extracting a list of observation files from Jedi config file")
+        observations = find_value_in_nested_dict(self.config, 'observations')
```

Comment:
Since `self.config` is the only thing used, perhaps we can turn this into a staticmethod, by passing the jedi config explicitly here.
```suggestion
        observations = find_value_in_nested_dict(jedi_config, 'observations')
```